RecoTracker/IterativeTracking/python/DisplacedRegionalStep_cff.py

```diff
@@ -0,0 +1,384 @@
+for _eraName, _postfix, _era in _cfg.nonDefaultEras():
+    _era.toReplaceWith(displacedRegionalStepClusters, _cfg.clusterRemoverForIter("DisplacedRegionalStep", _eraName, _postfix))
+
+# TRIPLET SEEDING LAYERS
+from RecoLocalTracker.SiStripClusterizer.SiStripClusterChargeCut_cfi import *
+import RecoTracker.TkSeedingLayers.seedingLayersEDProducer_cfi as _mod
+
+displacedRegionalStepSeedLayersTripl = _mod.seedingLayersEDProducer.clone(
+    layerList = [
+    #TOB
+    'TOB1+TOB2+MTOB3',
+    #TOB+MTEC
+    'TOB1+TOB2+MTEC1_pos','TOB1+TOB2+MTEC1_neg',
+    ],
+    TOB = dict(
+         TTRHBuilder      = cms.string('WithTrackAngle'),
+         clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutTight')),
+         matchedRecHits   = cms.InputTag('siStripMatchedRecHits','matchedRecHit'),
+         skipClusters     = cms.InputTag('displacedRegionalStepClusters')
+    ),
+    MTOB = dict(
+         TTRHBuilder      = cms.string('WithTrackAngle'),
+         clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutTight')),
+         skipClusters     = cms.InputTag('displacedRegionalStepClusters'),
+         rphiRecHits      = cms.InputTag('siStripMatchedRecHits','rphiRecHit')
+    ),
+    MTEC = dict(
+        rphiRecHits      = cms.InputTag('siStripMatchedRecHits','rphiRecHit'),
+        skipClusters     = cms.InputTag('displacedRegionalStepClusters'),
+        useRingSlector   = cms.bool(True),
+        TTRHBuilder      = cms.string('WithTrackAngle'),
+        clusterChargeCut = cms.PSet(refToPSet_ = cms.string('SiStripClusterChargeCutTight')),
+        minRing          = cms.int32(6),
+        maxRing          = cms.int32(7)
+    )
+)
+
+# Triplet TrackingRegion
+from RecoTracker.FinalTrackSelectors.displacedRegionalStepInputTracks_cfi import displacedRegionalStepInputTracks
+from RecoVertex.V0Producer.generalV0Candidates_cfi import generalV0Candidates as _generalV0Candidates
+from RecoTracker.DisplacedRegionalTracking.displacedRegionProducer_cfi import displacedRegionProducer as _displacedRegionProducer
+displacedRegionalStepSeedingV0Candidates = _generalV0Candidates.clone(
+    trackRecoAlgorithm = "displacedRegionalStepInputTracks",
+    doLambdas = False,
+    doFit = False,
+    useRefTracks = False,
+    vtxDecayXYCut = 1.,
+    ssVtxDecayXYCut = 5.,
+    allowSS = True,
+    innerTkDCACut = 0.2,
+    allowWideAngleVtx = True,
+    mPiPiCut = 13000.,
+    cosThetaXYCut = 0.,
+    kShortMassCut = 13000.,
+)
+displacedRegionalStepSeedingVertices = _displacedRegionProducer.clone(
+    minRadius = 2.0,
+    discriminatorCut = 0.5,
+    trackClusters = ["displacedRegionalStepSeedingV0Candidates", "Kshort"],
+)
+
+from RecoTracker.TkTrackingRegions.globalTrackingRegionWithVertices_cfi import globalTrackingRegionWithVertices as _globalTrackingRegionWithVertices
+displacedRegionalStepTrackingRegionsTripl = _globalTrackingRegionWithVertices.clone(RegionPSet = dict(
+    originRadius = 1.0,
```

Comment:
would efficiency suffer much if the origin radius is reduced here to some smaller value? e.g. 0.1 or even 0.05.
Similar for the `fixedError`, or, perhaps even `useFixedError` parameter.
The reference vertices are already displaced.
Shouldn't we use the vertex? or is there a point to search for further displacement wrt those displaced vertices?

Reply:
I just checked what happens if `originRadius` is changed from 1.0 to 0.8 (so approximately halving the volume of the tracking regions). Using the best config with both TOB/TEC and TIB/TID seeds as described in https://github.com/cms-sw/cmssw/pull/40643#issuecomment-1460430501, this decreased the CPU time of the DisplacedRegionalStep from (4.8 &pm; 0.4)% to (4.6 &pm; 0.5)% of tracking-only RECO CPU time, and the change in the efficiency of the DisplacedRegionalStep is shown in the plot below. The efficiency drops a few points at high displacements.

I could try decreasing the radius even further, but the overall CPU time of the iteration doesn't seem to depend very strongly on the tracking region volume.

![eff_vs_pos](https://user-images.githubusercontent.com/5114977/228607293-0e8fdb3c-2052-4bcf-8bf2-de75a8ed68c9.png)

Reply:
(@slava77, I originally posted this prematurely, and deleted it. Sorry for the confusion. Here is the reply with a little more information).

I checked what happens if `originRadius` is changed from 1.0 to 0.8 (so approximately halving the volume of the tracking regions) and also to 0.1. Using the best config with both TOB/TEC and TIB/TID seeds as described in https://github.com/cms-sw/cmssw/pull/40643#issuecomment-1460430501, this has the following effect on the CPU time of the DisplacedRegionalStep:

| `originRadius`  | fraction of tracking-only RECO CPU time |
| ------------- | ------------- |
| 1.0  | (4.8 &pm; 0.4)%  |
| 0.8 | (4.6 &pm; 0.5)% |
| 0.1 | (3.4 &pm; 0.3)% |

The change in the efficiency of the DisplacedRegionalStep for signal is shown in the plot below. There's not as big a decrease as I would expect going to 0.1, considering the tracking volumes are a thousand times smaller, although it is worse at very high displacements.

![eff_vs_pos](https://user-images.githubusercontent.com/5114977/235006424-642baf14-c646-423c-b46b-0ddb2dd7bca6.png)